### ISSUE TITLE
[onert] Introduce Mean op for training

### ISSUE THREAD

**Issue by jyoungyun** [Samsung/ONE#12357]
## What

Let's introduce Mean gradient kernel for training.

For #12325 

Draft #12358 

## Task

- [x] #12359
- [x] #12380
- [x] #12361

**Comment by jyoungyun** [comment]
### Test result

### test result

- Same loss

tensorflow
```
$ python3 tensorflow_run.py -m ./mnist_keras/mnist_keras_mean/ -i ../data/fashion_mnist/train.input.1000.bin -l ../data/fashion_mnist/train.output.1000.bin --data_length 1000 --epoch 5 --loss cce --optimizer adam  --batch_size 100 --learning_rat 0.001
Model: "sequential"
_________________________________________________________________
 Layer (type)                Output Shape              Param #
=================================================================
 conv2d (Conv2D)             (None, 28, 28, 32)        832

 global_average_pooling2d (  (None, 32)                0
 GlobalAveragePooling2D)

 flatten (Flatten)           (None, 32)                0

 dense (Dense)               (None, 10)                330

=================================================================
Total params: 1162 (4.54 KB)
Trainable params: 1162 (4.54 KB)
Non-trainable params: 0 (0.00 Byte)
_________________________________________________________________
Epoch 1/5
10/10 [==============================] - 1s 7ms/step - loss: 4.3883
Epoch 2/5
10/10 [==============================] - 0s 3ms/step - loss: 3.8929
Epoch 3/5
10/10 [==============================] - 0s 3ms/step - loss: 3.4905
Epoch 4/5
10/10 [==============================] - 0s 4ms/step - loss: 3.1690
Epoch 5/5
10/10 [==============================] - 0s 3ms/step - loss: 2.9193
==========================
Total time: 1.6967
```

onert_train (debug)
```
$ ./Product/out/bin/onert_train --modelfile ../Personal/models/mnist/mnist_keras/mnist_keras_mean/mnist_keras.circle --load_input:raw ../Personal/models/data/fashion_mnist/train.input.1000.bin --load_expected:raw ../Personal/models/data/fashion_mnist/train.output.1000.bin --loss 1 --optimizer 1 --epoch 5 --batch_size 100 --learning_rate 0.001
Model Expected Filename ../Personal/models/data/fashion_mnist/train.output.1000.bin
Model Input Filename ../Personal/models/data/fashion_mnist/train.input.1000.bin
Model Filename ../Personal/models/mnist/mnist_keras/mnist_keras_mean/mnist_keras.circle
== training parameter ==
- learning_rate   = 0.001
- batch_size      = 100
- loss_info       = {loss = categorical crossentropy, reduction = use default setting}
- optimizer       = adam
========================
Epoch 1/5 - time: 329.884ms/step - loss: [0] 4.3883
Epoch 2/5 - time: 334.767ms/step - loss: [0] 3.8929
Epoch 3/5 - time: 333.595ms/step - loss: [0] 3.4905
Epoch 4/5 - time: 337.305ms/step - loss: [0] 3.1690
Epoch 5/5 - time: 332.015ms/step - loss: [0] 2.9193
===================================
MODEL_LOAD   takes 1.0430 ms
PREPARE      takes 25.3870 ms
EXECUTE      takes 16682.5530 ms
- Epoch 1      takes 3298.8370 ms
- Epoch 2      takes 3347.6730 ms
- Epoch 3      takes 3335.9470 ms
- Epoch 4      takes 3373.0470 ms
- Epoch 5      takes 3320.1540 ms
===================================
```

onert_train (release)
```
$ ./Product/out/bin/onert_train --modelfile ../Personal/models/mnist/mnist_keras/mnist_keras_mean/mnist_keras.circle --load_input:raw ../Personal/models/data/fashion_mnist/train.input.1000.bin --load_expected:raw ../Personal/models/data/fashion_mnist/train.output.1000.bin --loss 1 --optimizer 1 --epoch 5 --batch_size 100 --learning_rate 0.001
Model Expected Filename ../Personal/models/data/fashion_mnist/train.output.1000.bin
Model Input Filename ../Personal/models/data/fashion_mnist/train.input.1000.bin
Model Filename ../Personal/models/mnist/mnist_keras/mnist_keras_mean/mnist_keras.circle
== training parameter ==
- learning_rate   = 0.001
- batch_size      = 100
- loss_info       = {loss = categorical crossentropy, reduction = use default setting}
- optimizer       = adam
========================
Epoch 1/5 - time: 41.574ms/step - loss: [0] 4.3883
Epoch 2/5 - time: 43.366ms/step - loss: [0] 3.8929
Epoch 3/5 - time: 44.951ms/step - loss: [0] 3.4905
Epoch 4/5 - time: 43.317ms/step - loss: [0] 3.1690
Epoch 5/5 - time: 43.205ms/step - loss: [0] 2.9193
===================================
MODEL_LOAD   takes 0.3930 ms
PREPARE      takes 12.3780 ms
EXECUTE      takes 2168.2970 ms
- Epoch 1      takes 415.7390 ms
- Epoch 2      takes 433.6600 ms
- Epoch 3      takes 449.5060 ms
- Epoch 4      takes 433.1680 ms
- Epoch 5      takes 432.0520 ms
```

**Comment by jyoungyun** [comment]
Done!
